arch/posix/CMakeLists.txt

```diff
@@ -133,7 +133,7 @@ if (CONFIG_GPROF)
   target_link_options(native_simulator INTERFACE "-pg")
 endif()
 
-zephyr_compile_definitions(_POSIX_C_SOURCE=200809 _XOPEN_SOURCE=600 _XOPEN_SOURCE_EXTENDED)
+zephyr_compile_definitions(_POSIX_C_SOURCE=200809L _XOPEN_SOURCE=600 _XOPEN_SOURCE_EXTENDED)
```

Comment:
This is still unfortunately a problem - even though the correct value of `_POSIX_C_SOURCE` is defined, it still sets the cflag globally, which can interfere with local cflags.

Reply:
There is another PR fixing this. It does not need to be done in one go.